core/src/main/scala/kafka/server/ClientRequestQuotaManager.scala

```diff
@@ -41,7 +41,7 @@ class ClientRequestQuotaManager(private val config: ClientQuotaManagerConfig,
     }
 
     if (quotasEnabled) {
-      val quotaSensors = getOrCreateQuotaSensors(request.session.sanitizedUser, request.header.clientId)
+      val quotaSensors = getOrCreateQuotaSensors(request.session.principal.getName, request.header.clientId)
```

Comment:
@ewencp  I think this is an inadvertent change since we are not changing the way usernames are handled? This is the reason for one of the test failures.